app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java

```diff
@@ -49,6 +49,7 @@
                 this.serviceLocator.postOnEventBus(new NewDataReceivedEvent(object.getData(), object.getClazz()));
             } catch (final ClassNotFoundException | IOException e) {
                 LOGGER.error(" Couldn't start deserialization!", e);
+                throw new DeserializationException(e);
```

Comment:
Are you 100% sure this is the only place where a `DeserializationException` should be thrown? It seems oddly specific for such a generic exception...